packages/jest-resolve/src/defaultResolver.ts

```diff
@@ -110,9 +110,7 @@ const defaultResolver: SyncResolver = (path, options) => {
     realpathSync,
   };
 
-  const pathToResolve = getPathInModule(path, resolveOptions);
-
-  const result = resolveSync(pathToResolve, resolveOptions);
+  const result = resolveByPathInModule(path, resolveOptions);
 
```

Comment:
I have to combine this to support array pattern.

Reply:
array is only supported in `imports`, right? would be nice to contain the change in just that code path

Reply:
if it's for both types, I'd prefer to migrate to your package in one PR, then add support for array patterns in a second PR

Reply:
> array is only supported in imports

Array is actually supported in both `imports` and `exports`. Just that `resolve.exports` does not support it.
https://github.com/lukeed/resolve.exports/issues/17

Reply:
Ah, ok. I'd still prefer adding support for that in a separate pr, though 🙂

Reply:
Actually after digging deeper into the rabbit hole, it seems like the array pattern is not used like the one described above.

https://github.com/nodejs/node/issues/44282

It seems to be for conditional imports:

```js
{
  "imports": {
    "#a": [{ "require": "./a.js" }, "./b.js"]
  }
}
```

Meaning that should be another assertion. Will clarify that on the Node.js side first.

In Node.js implementation:

- returns `null` when the array is empty
- returns `null` when an array entry is `null`
- returns `undefined` when the entry is a nested map `{ "a": { "b": "./c.js" } }`.

I think the last one is a bug.

Reply:
Code will be reverted